assets/src/edit-story/components/canvas/page.js

```diff
@@ -39,9 +40,16 @@ const Element = styled.div`
 `;
 
 function Page() {
+	const selectionRef = createRef();
+	const [ targetEl, setTargetEl ] = useState( null );
+
+	useEffect( () => {
+		setTargetEl( selectionRef.current );
+	}, [ selectionRef ] );
```

Comment:
`selectionRef` is guaranteed to never change. Should probably be `[selectionRef.current]` instead? Or ref-callback I mentioned above?

Reply:
Since `selectionRef` is an object then it would probably always be considered as not equal and always run. Wouldn't `selectionRef.current` always be `null` at this point instead?

Reply:
The result of `useRef()`  is guaranteed to always be the same object. E.g. `Object.is(old, new) === true`. If we use `selectionRef.current` it will first start as `null` and then will re-update itself with the actual element value. You'd have one extra render in that case, which is not ideal. A callback-ref could thus be better...

Reply:
Doesn't `Object.is( old, new )` always result in `false` since the references are compared here not the actual values of the two objects? E.g. if you do `a = createRef(); b = createRef()`, then `Object.is( a, b )` is always `false`.

(Note: just tested out with `selectionRef.current` and it doesn't seem to update -- the `targetEl` always stays `null` and `useEffect` is run just once. Perhaps I'm missing something & you tried it out and it worked for you?)

Reply:
With `[selectionRef]` the `useEffect` will definitely run only once since it never changes. In `a = useRef(); b = useRef()`, the `Object.is(a, b)` would definitely be `false`. But the `a = useRef()` in the same position will always return the same `a` object for any number of render calls. I'll try it now and see what's happening there. Possibly I'm missing something in the code. Does it work for you for the `selectionRef`?

Reply:
Yes, it does work with `selectionRef`.

Reply:
I just realized that the code uses `createRef`. I thought it was `useRef` hook instead? `createRef` indeed would create the new reference for each render. But that's not what you want here, is it? With hooks a `createRef` is no different that just a simple `let/const` var?

Reply:
Okay, just realized that the code is using `createRef` instead of `useRef` which seems to be creating a new object on each render.

Reply:
I tried with `useRef` just now and it all seems to work pretty well. Also I tried to just using `<Selection ref={setTargetEl}>` - that also works well. I guess either way works well.

Reply:
The last commit changed the code to use `<Selection ref={ setTargetEl }>` instead of `useEffect`. Thank you for going into such detail in reviewing btw!

Will merge this as a basis for future iterations.

Reply:
Thanks! I'm just trying to learn codebase this way :)